### PR TITLE
Fix test regression with enable-fips and enable-weak-ssl-ciphers

### PR DESCRIPTION
**Pull request by t8m**

### CODE REVIEW

**Comment by t8m** [comment]
This is rather a proof of concept than a real fix. I assume the real fix would be to add a gettable param to the 3DES in FIPS_MODULE that would indicate that it is a decrypt-only cipher.

**Comment by t8m** [comment]
@slontis @paulidale - we need to do something with this.

**Comment by slontis** [comment]
On a related note should we really be cross testing a PR against master, that seems like a strange combination.
Which seems equivalent to testing an older branch with the newest provider on that branch (Not sure why you would do that).

**Comment by slontis** [comment]
Also what is with these cross testing PR against branch results. Were they just recently added? There have been multiple times where issues are not seen until the next merged PR after the PR that caused the issues...

**Comment by paulidale** [comment]
The cross testing has been around for a while.  It's been update recently and made easier to run against PRs.

**Review by nhorman — APPROVED** [review]

File: ssl/ssl_lib.c [hunk 1]
```diff
@@ -7235,6 +7236,20 @@ const EVP_CIPHER *ssl_evp_cipher_fetch(OSSL_LIB_CTX *libctx,
      */
     ERR_set_mark();
     ciph = EVP_CIPHER_fetch(libctx, OBJ_nid2sn(nid), properties);
+    if (ciph != NULL) {
+        OSSL_PARAM params[2];
+        int decrypt_only = 0;
+
+        params[0] = OSSL_PARAM_construct_int(OSSL_CIPHER_PARAM_DECRYPT_ONLY,
```
Comment (mattcaswell) [inline ssl/ssl_lib.c:7243]: There needs to be some documentation about this new parameter

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: ssl/ssl_lib.c (hunk 1 above)
Comment (t8m) [reply ssl/ssl_lib.c:7243]: Good point, fixup pushed.

**Comment by t8m** [comment]
@nhorman please reconfirm

**Comment by nhorman** [comment]
Ack, approval holds

**Review by slontis — COMMENTED** [review]
An alternative to this would be using a fipsmodule.cnf that allows tdes encrypt?

**Comment by t8m** [comment]
> An alternative to this would be using a fipsmodule.cnf that allows tdes encrypt?

No, not really. We really should handle this case gracefully in libssl and not use a cipher that is unusable.

**Comment by t8m** [comment]
@mattcaswell please re-review

**Comment by t8m** [comment]
Merged to the master branch. Thank you for the reviews.

**Pull request closed without merging**